libs/fabric/src/lib/components/core/shared/changeable-helper.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { EventEmitter, QueryList } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { TypedChanges } from '../../../declarations/angular/typed-changes';
+import { ItemChangedPayload } from '../../core/declarative/item-changed.payload';
+import { ChangeableItemDirective } from './changeable-item.directive';
+
+/**
+ * Helper class for single changeable item
+ */
+export class ChangeableItemHelper<TItem> {
+  readonly onItemChanged = new EventEmitter<ItemChangedPayload<string, TItem>>();
+
+  constructor(private readonly key: string) {}
+
+  onChanges(changes: TypedChanges<TItem>) {
+    this.onItemChanged.emit({ key: this.key, changes });
+  }
+}
+
+/**
+ * Parent class for wrapper directive for multiple ChangeableItemDirectives
+ */
+export class ChangeableItemsHelper<TItem> {
+  readonly onChildItemChanged = new EventEmitter<ItemChangedPayload<string, TItem>>();
+  readonly onItemsChanged = new EventEmitter<QueryList<ChangeableItemDirective<TItem>>>();
+
+  private readonly _subscriptionsMap: { [key: string]: Subscription } = {};
+  private _changeSubscription: Subscription;
+
+  constructor(
+    private directiveItems: QueryList<ChangeableItemDirective<TItem>>,
+    private self?: IChangeableItemsContainer<TItem>,
+    private nonSelfHandler?: (nonSelfDirectives: ChangeableItemDirective<TItem>[]) => void
+  ) {}
+
+  afterContentInit() {
+    this._subscribeNewDirectives();
+    this._changeSubscription = this.directiveItems.changes.subscribe(newValues => {
+      this.onItemsChanged.emit(newValues);
+      this._subscribeNewDirectives();
+    });
+  }
+
+  onDestroy() {
+    Object.keys(this._subscriptionsMap).forEach(key => this._subscriptionsMap[key].unsubscribe());
```

Comment:
You can just use `Object.values`, since you don't need the key anyway.
Also, if you would need it - you can use `Object.entries` to get an array of `[key, value]` tuples.